File: pindeep_ios/PindeepSDK/PindeepClient.swift
import Foundation

class PindeepClient {
    let apiKey: String
    let baseUrl: String

    init(apiKey: String, baseUrl: String = "https://pindeep.link/api") {
        self.apiKey = apiKey
        self.baseUrl = baseUrl.hasSuffix("/") ? baseUrl : baseUrl + "/"
    }

    func createLink(data: [String: Any], completion: @escaping (Result<String, Error>) -> Void) {
        guard let url = URL(string: baseUrl + "create.php") else { return }

        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.addValue(apiKey, forHTTPHeaderField: "X-API-KEY")
        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try? JSONSerialization.data(withJSONObject: data)

        URLSession.shared.dataTask(with: request) { data, _, error in
            if let error = error { completion(.failure(error)); return }
            guard let data = data,
                  let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any],
                  let shortLink = json["short_link"] as? String
            else {
                completion(.failure(NSError(domain: "InvalidResponse", code: 0)))
                return
            }
            completion(.success(shortLink))
        }.resume()
    }

    func getStats(code: String, completion: @escaping (Result<[String: Any], Error>) -> Void) {
        guard let url = URL(string: baseUrl + "stats.php?code=" + code) else { return }

        var request = URLRequest(url: url)
        request.addValue(apiKey, forHTTPHeaderField: "X-API-KEY")

        URLSession.shared.dataTask(with: request) { data, _, error in
            if let error = error { completion(.failure(error)); return }
            guard let data = data,
                  let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
            else {
                completion(.failure(NSError(domain: "InvalidResponse", code: 0)))
                return
            }
            completion(.success(json))
        }.resume()
    }
}

File: pindeep_android/src/main/java/com/pindeep/PindeepClient.java
package com.pindeep;

import okhttp3.*;
import org.json.JSONObject;
import java.io.IOException;

public class PindeepClient {
    private final String apiKey;
    private final String baseUrl;
    private final OkHttpClient client;

    public PindeepClient(String apiKey, String baseUrl) {
        this.apiKey = apiKey;
        this.baseUrl = baseUrl.endsWith("/") ? baseUrl : baseUrl + "/";
        this.client = new OkHttpClient();
    }

    public void createLink(JSONObject json, Callback callback) {
        RequestBody body = RequestBody.create(json.toString(), MediaType.get("application/json"));
        Request request = new Request.Builder()
                .url(baseUrl + "create.php")
                .addHeader("X-API-KEY", apiKey)
                .post(body)
                .build();
        client.newCall(request).enqueue(callback);
    }

    public void getStats(String code, Callback callback) {
        Request request = new Request.Builder()
                .url(baseUrl + "stats.php?code=" + code)
                .addHeader("X-API-KEY", apiKey)
                .build();
        client.newCall(request).enqueue(callback);
    }
}